Lib/argparse.py

```diff
@@ -553,7 +553,10 @@ def _format_action_invocation(self, action):
                 default = self._get_default_metavar_for_optional(action)
                 args_string = self._format_args(action, default)
                 for option_string in action.option_strings:
-                    parts.append('%s %s' % (option_string, args_string))
+                    if args_string:
+                        parts.append('%s %s' % (option_string, args_string))
+                    else:
+                        parts.append('%s' % (option_string,))
```

Comment:
If `option_string` is a string, `'%s' % (option_string,)` can be replaced with just `option_string`. And the loop can be replaced with the single `extend()` call as at line 548.

But I'm not convinced that supporting empty names is a good idea.

Reply:
Hi @csabella, I'm looking at old PRs and this one seems still relevant as thee bug is still present on master. I think `parts.extend(action.option_strings)` was what Serhiy Stochaka meant by replacing the loop with `extend()`.